src/encoded/static/components/view_controls.js

```diff
@@ -1,105 +1,63 @@
-import _ from 'underscore';
 import * as encoding from '../libs/query_encoding';
 import { Modal, ModalHeader, ModalBody, ModalFooter } from '../libs/ui/modal';
 import { svgIcon } from '../libs/svg-icons';
 
 
 /**
- * Specifies the possible types of view controls. Extend this object with new view-control types
- * on the rare occasions that a new search view gets implemented. The string value matches the
- * @type of the search-view JSON. This object is visible externally to this module, so enforce its
- * immutability.
+ * Maps the page type to the corresponding view control elements. Extend this array on the rare
+ * occasions that a new search view gets implemented.
+ *   type: identifier for each element
+ *   path: goes between slashes in search path /{path}/
+ *   icon: variable name of icon in svg-icons.js for icon in button
+ *   title: Used in button; matrix prepended by type
+ *   label: Used for screen readers
  */
-export const ViewControlTypes = {
-    SEARCH: 'Search',
-    REPORT: 'Report',
-    MATRIX: 'Matrix',
-    SUMMARY: 'Summary',
-};
-Object.freeze(ViewControlTypes);
+const viewTypeMap = [
+    { type: 'search', path: 'search', icon: 'search', title: 'List', label: 'View search results as a list' },
+    { type: 'report', path: 'report', icon: 'table', title: 'Report', label: 'View search results as a tabular report' },
+    { type: 'matrix', path: 'matrix', icon: 'matrix', title: 'matrix', label: 'View search results as a matrix' },
+];
 
 
 /**
- * Maps search result @type[0] to corresponding view control path link components. Extend this
- * object with new pairs on the rare occasions that a new search view gets implemented. The keys
- * correspond to the values in `ViewControlTypes`, and the values fit in the view-control link
- * paths, e.g. the Matrix value, 'matrix', would fit between the slashes in the path "/matrix/" for
- * the matrix view control button.
+ * Based on the search-result context, determine which of the three types of search-result pages
+ * the user currently views. For this to work, matrix JSON has to have a `matrix` property at the
+ * top level of the JSON. If no criteria gets met, this returns an empty string. The "Summary"
+ * entry exists because the summary page shows the view controls though no view controls link to
+ * the summary page.
+ * @param {object} context Object currently in view.
+ *
+ * @return {string} Type of search page currently in view: list, report, matrix, summary.
  */
-const viewPathMap = {
-    Search: { path: 'search', icon: 'search', label: 'View results as a list' },
-    Report: { path: 'report', icon: 'table', label: 'View tabular report' },
-    Matrix: { path: 'matrix', icon: 'matrix', label: 'View summary matrix' },
-    Summary: { path: 'summary', icon: 'summary', label: 'View summary report' },
+const getSearchPageType = (context) => {
+    const atType = context['@type'][0];
+    if (atType === 'Search') {
+        return 'search';
```

Comment:
Should this and other result-values be constants?

Reply:
Yeah, that would be a good idea.